handlers/secrets.go

```diff
@@ -82,7 +82,17 @@ func UpdateSecrets(request requests.CreateFunctionRequest, deployment *v1beta1.D
 			},
 		}
 
-		deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, projectedSecrets)
+		existingVolumes := deployment.Spec.Template.Spec.Volumes
+		existingVolumeIndex := -1
+		for i, v := range existingVolumes {
+			if v.Name == volumeName {
+				existingVolumeIndex = i
+			}
+		}
+		if existingVolumeIndex > -1 {
+			existingVolumes = append(existingVolumes[:existingVolumeIndex], existingVolumes[existingVolumeIndex+1:]...)
```

Comment:
I think this can be moved to line number 89.

Reply:
I thought is was a faux-pas to edit the slice as your are iterating on it, but the tests still pass after I adjust it, so it doesn't seem to break.

Reply:
I need to find 10 minutes to sit down and review this in detail.